UltraStar Play/Assets/Common/Model/SettingsManager.cs

```diff
@@ -93,17 +94,54 @@ private Settings CreateDefaultSettings()
         if (!Application.isEditor)
         {
             // Create internal song folder on Android and add it to the settings.
-            string internalSongFolder = AndroidUtils.GetAppSpecificStorageAbsolutePath(false) + "/Songs";
-            if (!Directory.Exists(internalSongFolder))
+            try
             {
-                Directory.CreateDirectory(internalSongFolder);
+                string internalSongFolder = AndroidUtils.GetAppSpecificStorageAbsolutePath(false) + "/Songs";
+                if (!Directory.Exists(internalSongFolder))
+                {
+                    Directory.CreateDirectory(internalSongFolder);
+                }
+
+                defaultSettings.GameSettings.songDirs.Add(internalSongFolder);
+            }
+            catch (Exception ex)
+            {
+                Debug.LogError("Failed to create initial song folder.");
+                Debug.LogError(ex);
             }
-            defaultSettings.GameSettings.songDirs.Add(internalSongFolder);
         }
 #endif
+
+        // Try to select the first mic for singing.
+        try
+        {
+            MicProfile defaultMicProfile = CreateDefaultMicProfile();
+            if (defaultMicProfile != null)
+            {
+                settings.MicProfiles.Add(defaultMicProfile);
+            }
+        }
+        catch (Exception ex)
+        {
+            Debug.LogError("Failed to create initial recording device profile.");
+            Debug.LogError(ex);
+        }
+
         return defaultSettings;
     }
 
+    private MicProfile CreateDefaultMicProfile()
+    {
+        if (Microphone.devices.Length <= 0)
+        {
+            return null;
```

Comment:
Might be worth it to add some logging here, or maybe where ever the CreateDefaultSettings() is called.